### PR TITLE
Added tutorial showing how to access restrict a method in Sophia

### PR DESCRIPTION
**Pull request by Perseverance**

### CODE REVIEW

File: owning-a-smart-contract.md [hunk 1]
```diff
@@ -0,0 +1,423 @@
+# TUTORIAL: Sophia basics - owning a smart contract and owner privileges
+
+## Tutorial Overview
+This tutorial will walk teach you how one can access restrict a method to the deployer of the smart contract itself. This is a useful tool for defining critical business logic.
+## Prerequisites
+- Installed Aeproject - For installation steps reffer to [this tutorial](https://hackmd.aepps.com/MwMwrATA7AhjAsBaADARgGwkfKAOApogEYAm4xAnMlBBCTOsiCUA?view)
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: can you adjust the reference to be the file here in github?

File: owning-a-smart-contract.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 ## Tutorial Overview
 This tutorial will walk teach you how one can access restrict a method to the deployer of the smart contract itself. This is a useful tool for defining critical business logic.
 ## Prerequisites
-- Installed Aeproject - For installation steps reffer to [this tutorial](https://hackmd.aepps.com/MwMwrATA7AhjAsBaADARgGwkfKAOApogEYAm4xAnMlBBCTOsiCUA?view)
+- Installed Aeproject - For installation steps reffer to [this tutorial](https://github.com/aeternity/tutorials/blob/master/smart-contract-deployment-in-aeproject.md)
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: just `smart-contract-deployment-in-aeproject.md` is fine, then the pages will pick it up correctly

File: owning-a-smart-contract.md [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+mkdir ownable-project
+cd ownable-project
+aeproject init
+```
+
+### 1.2 aeproject epoch
+Another thing we would require is a local working environment to develop against. Luckily `aeproject` gives us that out of the box. Just type in the following command
+
+```
+aeproject node
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: when I install the aeproject from the default `develop` branch, I still don't have the `node` command

Comment (thepiwo) [reply owning-a-smart-contract.md]: Also to use the command, `docker` and `docker-compose` need to be installed, which is not explained in the referenced tutorial https://github.com/aeternity/tutorials/blob/5754bd471e7d5a283fbe99a8d2d503076701f813/smart-contract-deployment-in-aeproject.md

File: owning-a-smart-contract.md [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+Let's create a new empty folder and initialise an aepp project.
+
+```
+mkdir ownable-project
+cd ownable-project
+aeproject init
+```
+
+### 1.2 aeproject epoch
+Another thing we would require is a local working environment to develop against. Luckily `aeproject` gives us that out of the box. Just type in the following command
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: remove emphasizing such as "Luckily", this reads more like advertisement.

more explaination about what the command does should be included.

File: owning-a-smart-contract.md [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+
+### 2.1 Write the smart contract
+Lets first write our `Restricted.aes` smart contract.
+
+First create the contract file under contracts folder
+```
+touch ./contracts/Restricted.aes
+```
+
+The following code goes inside this file:
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: after using `touch` previously, mention any editor can be used to open the file

File: owning-a-smart-contract.md [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+
+```javascript=
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: I think usually 2 spaces are used for one indentation

Comment (thepiwo) [reply owning-a-smart-contract.md]: Also spaces before and after each `:` type definition are common in the core-devs sophia style

File: owning-a-smart-contract.md [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+As you can see this contract exposes two functions - both returning the number it has received as argument. Note that the restricted function is **not** actually restricted yet.
+### Write unit tests to call the methods
+In order to show the functionality we want we should write some unit tests.
+
+The main three unit tests needed are:
+1. The smart contract is being compiled and deployed successfully
+2. The non-restricted method is freely called by non-owner (more on owners soon)
+3. The restricted method cannot be called by non-owner
+
+#### The code
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: where to put it?

File: owning-a-smart-contract.md [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+```
+
+## Access restrict the smart contract
+
+In order to restrict a method to be called only by the deployer of the smart contract we would need to modify the smart contract to know who is their deployer. Here is how to do that
+
+### Modify the smart contract
+
+Let's first add state variable for the owner and set it on deploy time. On the second line of Restricted.aes put the following:
+
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: I think code style should be more like

```
 contract Restricted =
  record state = 
    { owner : address}

  public stateful function init() = 
    { owner = Call.caller } // Initializing the owner to the deployer
```

File: owning-a-smart-contract.md [hunk 9]
```diff
@@ -0,0 +1,423 @@
...
+})
+```
+
+As we have not yet implemented the restriction functionality the third test naturally fails.
+Run the following command in the terminal:
+```
+aeproject test
+```
+
+```cmake=
```
Comment (thepiwo) [inline owning-a-smart-contract.md]: when I replace the `exampleTest.js` with it  the tests fail:

```
===== Starting Tests =====


  Restricted
    1) "before all" hook


  0 passing (3ms)
  1 failing

  1) Restricted
       "before all" hook:
     ReferenceError: wallets is not defined
      at Context.before (test/exampleTest.js:23:16)
      at process.topLevelDomainCallback (domain.js:120:23)
```

File: owning-a-smart-contract.md (hunk 3 above)
Comment (Perseverance) [reply owning-a-smart-contract.md]: aeproject PR is now merged

File: owning-a-smart-contract.md (hunk 9 above)
Comment (Perseverance) [reply owning-a-smart-contract.md]: aeproject PR is now merged

**Pull request merged**